### PR TITLE
[logstash bridge]: stableapi bootstrap

### PR DESCRIPTION
**Pull request by yaauie**
Adds a new `logstash-bridge` project in `/libs` that exports api-stable wrappers for the elasticsearch-internal types that Logstash's Elastic Integration Filter relies on to provide ingest pipeline execution inside of Logstash.

This is an alternate approach to #107973 is shaped by feedback on the other approach.

The resulting project is _not_ intended for distribution, but as a build-time dependency of the separately-packaged Logstash Integration Filter.

## TODO:

 - [x] map codeowners entry to public team
 - [x] determine if we can limit the scope of exports in `module-info.java`
 - ~~[ ] determine feasibility of producing an uberjar artifact with the transitive dependencies (note: still would not be _published_, as it could be built from the downstream project)~~

### CODE REVIEW

File: libs/logstash-bridge/src/main/java/org/elasticsearch/logstashbridge/common/Settings.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.logstashbridge.common;
+
+import org.elasticsearch.logstashbridge.StableAPI;
+
+public class Settings extends StableAPI.Proxy<org.elasticsearch.common.settings.Settings> {
```
Comment (rjernst) [inline libs/logstash-bridge/src/main/java/org/elasticsearch/logstashbridge/common/Settings.java:12]: Naming these the exact same as the classes in server presents problems for quickly opening the files in intellij. Can we please have all of the proxies with qualified names?

File: test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java [hunk 2]
```diff
@@ -222,6 +222,7 @@ static Map<String, URL> getCodebases() {
         addClassCodebase(codebases, "elasticsearch-cli", "org.elasticsearch.cli.Command");
         addClassCodebase(codebases, "elasticsearch-preallocate", "org.elasticsearch.preallocate.Preallocate");
         addClassCodebase(codebases, "elasticsearch-vec", "org.elasticsearch.vec.VectorScorer");
+        addClassCodebase(codebases, "elasticsearch-logstash-bridge", "org.elasticsearch.logstashbridge.LogstashBridge");
```
Comment (rjernst) [inline test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java:225]: This shouldn't be necessary? It only applies to tests, and when running with security manager, but the bridge shouldn't need ESTestCase or to run with security manager right?

File: libs/logstash-bridge/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## Logstash Bridge
+
+This package contains bridge functionality and API-stability tests to ensure that Logstash's Elastic Integration plugin has access to the minimal subset of Elasticsearch to perform its functions.
```
Comment (rjernst) [inline libs/logstash-bridge/README.md:3]: > API-stability tests

It's not tests, but the an actual stable bridge API right?

**Review by rjernst — CHANGES_REQUESTED** [review]
Thanks for the additional wrappers, I think they will help maintain the stable interface logstash is looking for.

My main remaining concern is about the naming. We often quickly navigate to classes in intellij, and have had problems in the past when classes of the same name exist in two different modules. I think these class names need to be unique, perhaps with a common prefix to indicate it's part of the logstash bridge.

File: test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java (hunk 2 above)
Comment (yaauie) [reply test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java:225]:
```suggestion
```

File: libs/logstash-bridge/src/main/java/org/elasticsearch/logstashbridge/StableAPI.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * A {@code StableAPI} is the stable form of an Elasticsearch API, and can produce comparable instances
+ * from the actual API that they mirror. As part of the LogstashBridge project, these classes are relied
+ * upon by the "Elastic Integration Filter Plugin" for Logstash and their external shapes mut not change
+ * without coordination with the maintainers of that project.
+ *
+ * @param <T> the actual type of the Elasticsearch API being mirrored
+ */
+public interface StableAPI<T> {
```
Comment (rjernst) [inline libs/logstash-bridge/src/main/java/org/elasticsearch/logstashbridge/StableAPI.java:23]: nit: let's name this StableBridgeAPI?

**Review by rjernst — APPROVED** [review]
Thanks for the renaming. One last nit on the base name, otherwise LGTM.

**Pull request merged**